Assignment3/Q1.py

```diff
@@ -0,0 +1,128 @@
+"""The approach taken in this solution is based on a DFS-like search. The worst case running time
```

Comment:
Actually, in a graph with V vertices and E edges, DFS is O(V+E).
In this case, if you have an NxN grid, what is the worst case complexity? Is it actually exponential?